src/qilib/configuration_helper/configuration_helper.py

```diff
@@ -0,0 +1,96 @@
+from qilib.utils.storage.interface import StorageInterface
+
+
+class ConfigurationHelper:
+    """  Provides access to the instrument configuration database and maintains two ConfigurationSets."""
+
+    def __init__(self, storage: StorageInterface,
+                 active_configuration: Union[InstrumentConfigurationSet, None] = None,
+                 inactive_configuration: Union[InstrumentConfigurationSet, None] = None) -> None:
+        """ Instantiate a new ConfigurationHelper.
+
+        Args:
+            storage: A StorageInterface that stores configurations.
+            active_configuration: An optional instrument configuration set.
+            inactive_configuration: An optional inactive instrument configuration set.
+        """
+        if active_configuration is not None:
+            self.active_configuration = active_configuration
+        else:
+            self.active_configuration = InstrumentConfigurationSet(storage)
+        if inactive_configuration is not None:
+            self.inactive_configuration = inactive_configuration
+        else:
+            self.inactive_configuration = InstrumentConfigurationSet(storage)
+        self._storage = storage
+
+    def snapshot(self, tag: List[str]) -> None:
+        """ Overwrite the tag and refreshes all underlying InstrumentConfiguration's on the active configuration.
+
+        Args:
+            tag: Unique identifier for an InstrumentConfigurationSet.
+
+        """
+        self.active_configuration.snapshot(tag)
+
+    def retrieve_inactive_configuration_from_storage(self, tag: List[str]) -> None:
+        """ Load configuration set from storage and set as inactive.
+
+        Args:
+            tag: Unique identifier for an InstrumentConfigurationSet.
+
+        """
+        self.inactive_configuration = InstrumentConfigurationSet.load(tag, self._storage)
+
+    def write_active_configuration_to_storage(self) -> None:
+        """ Write the active configuration set to storage."""
+        self.active_configuration.store()
+
+    def write_inactive_configuration_to_storage(self) -> None:
+        """ Write the inactive configuration set to storage."""
+        self.inactive_configuration.store()
+
+    def apply_inactive(self) -> None:
+        """ Apply inactive configuration and set as active."""
+        self.inactive_configuration.apply()
+        self.active_configuration = self.inactive_configuration
+
+    def apply_inactive_delta(self) -> None:
+        """ Call apply_delta on inactive configuration set and set it as active."""
+        self.inactive_configuration.apply_delta()
+        self.active_configuration = self.inactive_configuration
+
+    def apply_inactive_delta_lazy(self) -> None:
+        """ Call apply_delta_lazy on inactive configuration set and set it as active."""
+        self.inactive_configuration.apply_delta_lazy()
+        self.active_configuration = self.inactive_configuration
+
+    def get_tag_by_label(self, label: str) -> Any:
```

Comment:
Isn't return value type Any actually List[str]. You want to use to call retrieve_inactive_configuration_from_storage(..) as in:
a = get_tag_by_label('online')
retrieve_inactive_configuration_from_storage(a)
apply_inactive()

Reply:
I know, that is a problem here. The underlying storage interface can store, and retrieve, any type of serializable data. So if I say the return type here is List[str] mypy complains that the ` self._storage.load_data(['labels', label])` returns Any.

I did not see a quick fix, so as discussed, we should use `Any` instead of wasting time on typehint problems.